### ISSUE TITLE
gzipped body  cause json_encode error

### ISSUE THREAD

**Issue by rtyshyk** [getsentry/sentry-php#950]
Hi there,

One on of API receives compressed json (gzip), what cause issues to send request body to the sentry and app is crashing. The request contains proper headers, e.g.

```
Content-Type: application/json
Content-Encoding: gzip
```

Stack trace
```
"[2020-01-09 07:52:52] prod.ERROR: Could not encode value into JSON format. Error was: \"Malformed UTF-8 characters, possibly incorrectly encoded\". {\"exception\":\"[object] (Sentry\\\\Exception\\\\JsonException(code: 0): Could not encode value into JSON format. Error was: \\\"Malformed UTF-8 characters, possibly incorrectly encoded\\\". at \/app\/source\/vendor\/sentry\/sentry\/src\/Util\/JSON.php:28)",
"[stacktrace]",
"#0 \/app\/source\/vendor\/sentry\/sentry\/src\/Transport\/HttpTransport.php(112): Sentry\\\\Util\\\\JSON::encode(Object(Sentry\\\\Event))",
"#1 \/app\/source\/vendor\/sentry\/sentry\/src\/Client.php(120): Sentry\\\\Transport\\\\HttpTransport->send(Object(Sentry\\\\Event))",
"#2 \/app\/source\/vendor\/sentry\/sentry\/src\/State\/Hub.php(146): Sentry\\\\Client->captureEvent(Array, Object(Sentry\\\\State\\\\Scope))",
"#3 \/app\/source\/app\/Common\/Monolog\/SentryHandler.php(63): Sentry\\\\State\\\\Hub->captureEvent(Array)",
"#4 \/app\/source\/vendor\/sentry\/sentry\/src\/State\/Hub.php(87): App\\\\Common\\\\Monolog\\\\SentryHandler->App\\\\Common\\\\Monolog\\\\{closure}(Object(Sentry\\\\State\\\\Scope))",
"#5 \/app\/source\/app\/Common\/Monolog\/SentryHandler.php(64): Sentry\\\\State\\\\Hub->withScope(Object(Closure))",
"#6 \/app\/source\/vendor\/monolog\/monolog\/src\/Monolog\/Handler\/AbstractProcessingHandler.php(42): App\\\\Common\\\\Monolog\\\\SentryHandler->write(Array)",
"#7 \/app\/source\/vendor\/monolog\/monolog\/src\/Monolog\/Logger.php(323): Monolog\\\\Handler\\\\AbstractProcessingHandler->handle(Array)",
"#8 \/app\/source\/vendor\/monolog\/monolog\/src\/Monolog\/Logger.php(541): Monolog\\\\Logger->addRecord(400, 'Could not encod...', Array)",
"#9 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Log\/Logger.php(176): Monolog\\\\Logger->error('Could not encod...', Array)",
"#10 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Log\/Logger.php(87): Illuminate\\\\Log\\\\Logger->writeLog('error', 'Could not encod...', Array)",
"#11 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Log\/LogManager.php(547): Illuminate\\\\Log\\\\Logger->error('Could not encod...', Array)",
"#12 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Exceptions\/Handler.php(121): Illuminate\\\\Log\\\\LogManager->error('Could not encod...', Array)",
"#13 \/app\/source\/app\/Exceptions\/Handler.php(56): Illuminate\\\\Foundation\\\\Exceptions\\\\Handler->report(Object(Sentry\\\\Exception\\\\JsonException))",
"#14 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Routing\/Pipeline.php(49): App\\\\Exceptions\\\\Handler->report(Object(Sentry\\\\Exception\\\\JsonException))",
"#15 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php(176): Illuminate\\\\Routing\\\\Pipeline->handleException(Object(Illuminate\\\\Http\\\\Request), Object(Sentry\\\\Exception\\\\JsonException))",
"#16 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Pipeline\/Pipeline.php(105): Illuminate\\\\Pipeline\\\\Pipeline->Illuminate\\\\Pipeline\\\\{closure}(Object(Illuminate\\\\Http\\\\Request))",
"#17 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Kernel.php(145): Illuminate\\\\Pipeline\\\\Pipeline->then(Object(Closure))",
"#18 \/app\/source\/vendor\/laravel\/framework\/src\/Illuminate\/Foundation\/Http\/Kernel.php(110): Illuminate\\\\Foundation\\\\Http\\\\Kernel->sendRequestThroughRouter(Object(Illuminate\\\\Http\\\\Request))",
"#19 \/app\/source\/public\/index.php(52): Illuminate\\\\Foundation\\\\Http\\\\Kernel->handle(Object(Illuminate\\\\Http\\\\Request))",
"#20 {main}",
"\"} ",
```

Expected result: sentry decodes the request and the make it json. Or does not trying to create json from binary data.

**Comment by ste93cry** [comment]
This should have been solved in `2.2.6` (#934). Note that on PHP < `7.2` the encoding of data containing invalid UTF-8 characters is done in a best-effort way and could still fail. Instead from PHP >= `7.2` we use the `JSON_INVALID_UTF8_SUBSTITUTE` flag

**Comment by rtyshyk** [comment]
@ste93cry Looks fine now. No crash, but no body :)

Thanks!
